src/spaces/discrete.jl

```diff
@@ -75,10 +75,10 @@ function empty_positions(model::ABM{<:DiscreteSpace})
 end
 
 """
-    isempty(position, model::ABM{<:DiscreteSpace})
+    isempty(pos, model::ABM{<:DiscreteSpace})
 Return `true` if there are no agents in `position`.
 """
-Base.isempty(pos, model::ABM) = isempty(ids_in_position(pos, model))
+Base.isempty(pos::ValidPos, model::ABM{<:DiscreteSpace}) = isempty(ids_in_position(pos, model))
```

Comment:
All these functions that use space are probably better done if they dispatch on the space instance and then the space instance implements `isempty`. This way we don't have to re-code twice all functions that ultimately dispatch on space.

Reply:
By that I mean that `isempty` dispatches to `isempty(pos, abmspace(model))`.

Reply:
yes indeed this is an instance of #910

Reply:
actually it is not possible, the implementation use also the model (see e.g. `ids_in_position(pos, model)`) , but I actually would like to use the "trick" described in #910, but given it is a general refactoring maybe we can wait till that issue is solved

Reply:
How does `ids_in_position` use the model? It really shouldn't!

Reply:
yes, I know :D but there are many functions which should really belong to the space but need to keep a reference to the model, e.g. `random_id` since it needs the rng of the model.                                                                                                  

If we want to let the user access the space eventually for the multi-space models we should really follow a unique convention, namely passing the model as argument after the space, namely in this case `ids_in_position(pos, space, ::ABM)`. So my idea is that in the end this will be needed to be changed again, but probably it is ok to start dispatching now like this even if in the future it will be modified

Reply:
> All these functions that use space are probably better done if they dispatch on the space instance and then the space instance implements `isempty`. This way we don't have to re-code twice all functions that ultimately dispatch on space.

Actually it is not true we should then implement `ids_in_position(pos, space)` for all discrete spaces and keep `ids_in_position(pos, model)` for users. So I think not worth right now.

If you agree this PR should be good to go!